File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int s = 0, e = nums.length - 1;
    int mid = s + (e-s)/2;
    while(s<=e){
        if(nums[mid] == target){
            return mid;
        }
        else if(nums[mid] < target){
            s = mid + 1;
        }
        else if(nums[mid] > target){
            e = mid - 1;
        }
        mid = s + (e-s)/2;
    }
    return -1;
}
        
 
}

File: 704-binary-search/704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n=nums.size()-1;
        int low=0, mid;
        int high=n;
        while(low<=high)
        {
            mid=(low+high)/2;
        
            if(target==nums[mid])return mid;
            else if(nums[mid]>target)
            {
               high=mid-1;
            }
            else
            {
               low=mid+1;
            }
        }
        return -1;
        
        
    }
};